private/buf/cmd/buf/command/registry/registrylogin/registrylogin.go

```diff
@@ -128,6 +129,26 @@ func run(
 			return err
 		}
 	}
+	registryProvider, err := bufcli.NewRegistryProvider(ctx, container)
+	if err != nil {
+		return err
+	}
+	authnService, err := registryProvider.NewAuthnService(ctx, remote)
+	if err != nil {
+		return err
+	}
+	user, err := authnService.GetCurrentUser(rpcauth.WithToken(ctx, token))
+	if err != nil {
+		// We don't want to use the default error from wrapError here if the error
+		// an unauthenticated error.
+		return errors.New("invalid token provided")
+	}
+	if user == nil {
+		return errors.New("no user found for provided token")
+	}
+	if user.Username != username {
+		return fmt.Errorf("token user does not match username provided: %s", username)
```

Comment:
nit:
```suggestion
		return fmt.Errorf("the username associated with that token (%s) does not match the username provided (%s)", user.Username, username)
```

We _might_ want to let the user know what the resolved `user.Username` was so that it's clear what the difference is. At first I thought that might be leaking information, but the user can discover who's token it is by other means so this seems relatively harmless (but helpful).

Reply:
Yeah, that's a good point -- I went back and forth on this also, for the exact reason that you stated around leaking information. But yeah, you're right, ultimately this is something that can already be discovered (by basically make this exactly call, lol), so it's probably fine. Will update this and the `CHANGELOG` :D